app/src/main/java/com/keylesspalace/tusky/components/report/fragments/ReportStatusesFragment.kt

```diff
@@ -118,62 +118,46 @@ class ReportStatusesFragment : Fragment(R.layout.fragment_report_statuses), Inje
                 animateEmojis = preferences.getBoolean(PrefKeys.ANIMATE_CUSTOM_EMOJIS, false)
         )
 
-        adapter = StatusesAdapter(statusDisplayOptions,
-                viewModel.statusViewState, this)
+        adapter = StatusesAdapter(statusDisplayOptions, viewModel.statusViewState, this)
 
         binding.recyclerView.addItemDecoration(DividerItemDecoration(requireContext(), DividerItemDecoration.VERTICAL))
         binding.recyclerView.layoutManager = LinearLayoutManager(requireContext())
         binding.recyclerView.adapter = adapter
         (binding.recyclerView.itemAnimator as SimpleItemAnimator).supportsChangeAnimations = false
 
-        viewModel.statuses.observe(viewLifecycleOwner) {
-            adapter.submitList(it)
-        }
-
-        viewModel.networkStateAfter.observe(viewLifecycleOwner) {
-            if (it?.status == com.keylesspalace.tusky.util.Status.RUNNING)
-                binding.progressBarBottom.show()
-            else
-                binding.progressBarBottom.hide()
-
-            if (it?.status == com.keylesspalace.tusky.util.Status.FAILED)
-                showError(it.msg)
+        lifecycleScope.launch {
+            viewModel.statusesFlow.collectLatest { pagingData ->
+                adapter.submitData(pagingData)
+            }
         }
 
-        viewModel.networkStateBefore.observe(viewLifecycleOwner) {
-            if (it?.status == com.keylesspalace.tusky.util.Status.RUNNING)
-                binding.progressBarTop.show()
-            else
-                binding.progressBarTop.hide()
-
-            if (it?.status == com.keylesspalace.tusky.util.Status.FAILED)
-                showError(it.msg)
-        }
+        adapter.addLoadStateListener { loadState ->
+            (loadState.refresh as? LoadState.Error?
```

Comment:
I would probably do `if` instead, it's a bit obfuscated

Reply:
good point